Modules/Common/src/TrendCheck.cxx

```diff
@@ -0,0 +1,466 @@
+      break;
+  }
+
+  // number of points, excluding the last one, used to compute the average and standard deviation of the trend
+  // all points are considered if the value is smaller or equal to zero
+  parKey = "nPointsForAverage";
+  parOpt = mCustomParameters.atOptional(parKey, mActivity);
+  if (!parOpt) {
+    parOpt = mCustomParameters.atOptional(parKey);
+  }
+  if (parOpt.has_value()) {
+    if (parOpt.value() == "all") {
+      mNPointsForAverage = 0;
+    } else {
+      mNPointsForAverage = std::stoi(parOpt.value());
+    }
+  }
+
+  if (mNPointsForAverage == 0) {
+    ILOG(Info, Support) << "using all points for statistics calculation" << ENDM;
+  } else {
+    ILOG(Info, Support) << "using at most " << mNPointsForAverage << " points for statistics calculation" << ENDM;
+  }
+
+  // position and size of the label showing the result of the check
+  parKey = "qualityLabelPosition";
+  parOpt = mCustomParameters.atOptional(parKey, mActivity);
+  if (!parOpt) {
+    parOpt = mCustomParameters.atOptional(parKey);
+  }
+  if (parOpt.has_value()) {
+    auto position = o2::utils::Str::tokenize(parOpt.value(), ',', false, true);
+    if (position.size() == 2) {
+      mQualityLabelPosition = std::make_pair<float, float>(std::stof(position[0]), std::stof(position[1]));
+    }
+  }
+
+  parKey = "qualityLabelSize";
+  parOpt = mCustomParameters.atOptional(parKey, mActivity);
+  if (!parOpt) {
+    parOpt = mCustomParameters.atOptional(parKey);
+  }
+  if (parOpt.has_value()) {
+    auto position = o2::utils::Str::tokenize(parOpt.value(), ',', false, true);
+    if (position.size() == 2) {
+      mQualityLabelSize = std::make_pair<float, float>(std::stof(position[0]), std::stof(position[1]));
+    }
+  }
+}
+
+void TrendCheck::endOfActivity(const Activity& activity)
+{
+  // reset all members
+  mActivity = Activity{};
+  mAverageTrend.clear();
+  mThresholdsTrendBad.clear();
+  mThresholdsTrendMedium.clear();
+  mQualities.clear();
+}
+
+static std::string getBaseName(std::string name)
+{
+  auto pos = name.rfind("/");
+  return ((pos < std::string::npos) ? name.substr(pos + 1) : name);
+}
+
+// compute the mean and standard deviation from a given number of points bewfore the last one
+// all points are used if nPointsForAverage <= 0
+static std::optional<std::pair<double, double>> getGraphStatistics(TGraph* graph, int nPointsForAverage)
+{
+  std::optional<std::pair<double, double>> result;
+
+  int nPoints = graph->GetN();
+  // we need at least two points before the last one
+  if (nPoints < 3) {
+    return result;
+  }
+
+  // index of the last-but-one point
+  int pointIndexMax = nPoints - 2;
+  // index of the starting point for the mean and stadrad deviation computation
+  int pointIndexMin = (nPointsForAverage > 0 && pointIndexMax >= nPointsForAverage) ? pointIndexMax - nPointsForAverage + 1 : 0;
+  // number of points used for the computation, must be greater than one
+  int N = pointIndexMax - pointIndexMin + 1;
```

Comment:
lower case and if possible descriptive name

Reply:
Done.